src/runtimes/node/bundler.js

```diff
@@ -46,7 +35,25 @@ const bundleJsFile = async function ({
     }
   }
 
-  return { bundlePath, cleanTempFiles, data }
+  try {
+    const data = await buildFunction({
+      bundle: true,
+      entryPoints: [srcFile],
+      external,
+      logLevel: 'warning',
+      outfile: bundlePath,
+      nodePaths: additionalModulePaths,
+      platform: 'node',
+      resolveExtensions: ['.js', '.jsx', '.mjs', '.cjs', '.json'],
+      target: ['es2017'],
+    })
+
+    return { bundlePath, cleanTempFiles, data }
+  } catch (error) {
+    error.errorInfo = { type: 'functionsBundling' }
```

Comment:
At the moment, the `functionsBundling` error type has a required field `location.functionName` which is used in the error message.

https://github.com/netlify/build/blob/fb47c2059d7ad46c48df5d715de759de38c0712d/packages/build/src/error/type.js#L86

I.e. this would need to be `error.errorInfo = { type: 'functionsBundling', location: { functionName } }`.

Reply:
Ah, I didn't know this. Thanks for flagging! Done in https://github.com/netlify/zip-it-and-ship-it/pull/404/commits/920ee876b5874a8f4348e4ba15d9ff5451fcce8c.